test/shell-logging.test.ts

```diff
@@ -1,137 +1,102 @@
 import { mockSpawn } from './mock-child_process';
 import mockfs from './mock-fs';
+import { MockProgressListener } from './mock-progress-listener';
 import { setLogThreshold } from '../bin/logging';
-import { shell } from '../lib/private/shell';
+import { EventType } from '../lib';
+import { shell, ProcessFailedError } from '../lib/private/shell';
+
 jest.mock('child_process');
 
-describe('logging', () => {
-  let consoleSpy: jest.SpyInstance;
+describe('shell', () => {
+  let progressListener: MockProgressListener;
+  let eventPublisher: (type: EventType, message: string, forceStdOut?: boolean) => void;
 
   beforeEach(() => {
-    consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+    progressListener = new MockProgressListener();
```

Comment:
Ended up discovering a bug with the original testing set up. As it was just listening for console.error or console.log, the tests would rarely fail. This seemed like it was potentially the result of capturing output from other streams or somehow canceling the stream early. Regardless I swapped it to this design to ensure I was always getting it (also much easier to test in general)